### PR TITLE
Make the Organizations a default feature

### LINKED ISSUE #32395: Remove the organization feature and make the feature a default feature
We also need to update the release notes and blog post.

### PR DESCRIPTION
**Pull request by vramik**
Closes #32395

### CODE REVIEW

**Comment by sschu** [comment]
Hi! @vramik I am wondering what the general rationale is for having a features as a `Feature` or not.  I guess for the user profile, the rationale for removing it as a feature was that everybody should actually use the user profile so you cannot deactivate it. 

I think @vmuzikar mentioned in some other discussion that down the road, it would be good if we could use `Features` to completely remove code for things people don't need to reduce the attack surface. He used SAML as something people might not want to use at all. I can definitely see people not needing organizations, so I am wondering where the line is between features that can be disabled and core functionality which cannot.

**Comment by vramik** [comment]
@sschu It has been decided that organizations are meant to be enabled/disabled for each individual realm by its setting, therefore there is no need for specific organization feature, as a result we can remove all those checks in the code and make it more maintainable.

**Comment by vramik** [comment]
> I see the maintainability benefits, especially if the code is all over the place. But you lose the ability to eliminate the code if it is not a feature.
> You can also use/not use SAML on a per-realm basis. Still it might make sense to be able to eliminate it. For SAML this is more clear as the code is probably less cluttered and disabling it eliminates a whole set of endpoints. 

+1

> Still, I am wondering how we decide if something should be a feature or a core functionality.

TBH, I do not know exact process ... maybe @pedroigor, could you please comment on this one?

**Review by vmuzikar — COMMENTED** [review]
My understanding was that Profile Features will be heavily leveraged in the near future as it might be useful for tracking features lifecycle for users (it's clear by just looking at the docs or server info endpoint at what "life stage" is currently a given feature at), versioning of features in case we need to do breaking changes to them, and also the mentioned "code elimination" in the future.

Also I see removing a Profile Feature as maybe unnecessary breaking change.

The comment @sschu meant is probably [this](https://github.com/keycloak/keycloak/pull/32309#issuecomment-2312987421).

CC @stianst – you probably have more context on it than me.

**Comment by stianst** [comment]
As there is support for a feature flag that should remain; that allows someone that wants to disable organization support to disable it; and in addition we can introduce a new version of organizations if there is a need to make breaking changes in the future.

**Review by stianst — CHANGES_REQUESTED** [review]
It's perfectly fine to have organizations enabled by default; although maybe they shouldn't be, as most folks won't use them.

In either case those please leave the feature in place, and just make it fully supported.

**Review by pedroigor — DISMISSED** [review]

**Review by stianst — DISMISSED** [review]

File: docs/documentation/release_notes/topics/26_0_0.adoc [hunk 1]
```diff
@@ -104,6 +104,12 @@ when exporting a realm.
 
 For information on how to migrate, see the link:{upgradingguide_link}[{upgradingguide_name}].
 
+= Keycloak Organization is now a supported feature
+
+In previous version there have been introduced Keycloak Organizations as a technology preview feature.
+
+Since {project_name} 26 releate this feature becomes supported.
```
Comment (sguilhen) [inline docs/documentation/release_notes/topics/26_0_0.adoc:111]:
```suggestion
The organizations feature, introduced in Keycloak 25 as technology preview, is now fully supported.
```

**Review by sguilhen — APPROVED** [review]
LGTM @vramik  - left just a small suggestion for the release notes text

File: docs/documentation/release_notes/topics/26_0_0.adoc [hunk 2]
```diff
@@ -104,6 +104,10 @@ when exporting a realm.
 
 For information on how to migrate, see the link:{upgradingguide_link}[{upgradingguide_name}].
 
+= Keycloak Organization is now a supported feature
+
+The organizations feature, introduced in {project_name} 25 as technology preview, is now fully supported.
+
```
Comment (sguilhen) [inline docs/documentation/release_notes/topics/26_0_0.adoc:110]: Not sure we should use {project_name} here because we don't have a RHBK 25, do we?

Comment (vramik) [reply docs/documentation/release_notes/topics/26_0_0.adoc:110]: true, so I'll probably revert back, release notes are pulled into RHBK docs and we do not want to have `Keycloak` there, right?

**Review by sguilhen — APPROVED** [review]

**Review by ahus1 — APPROVED** [review]
Approving based on @sguilhen's review

**Pull request merged**